### PR TITLE
Power armor

### PR DESCRIPTION
**Pull request by DeadLineSS13**
## About The Pull Request
This PR advance PA system.

## Why It's Good For The Game
Now PA is small tank that can take damage. And module system make PA more diffrents PA styles.

## Changelog

:cl: DeadLineSS13
add: Parts PA
add: Wokbench for PA
refactor: PA integarity.
/:cl:

### CODE REVIEW

File: mojave/code/modules/clothing/spacesuits/power_armor_parts.dm [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+					CUTTING = 0, \
+					PIERCING = 0, \
+					IMPALING = 0, \
+					LASER = 0, \
+					ENERGY = 0, \
+					FIRE = 0)
+	uses_integrity = TRUE
+	max_integrity = 100
+	var/icon_state_pa
+	var/chanse = 0 //Weight for pick
```
Comment (ProfessorPopoff) [inline mojave/code/modules/clothing/spacesuits/power_armor_parts.dm]: Chanse should be "Chance"

File: mojave/code/modules/mob/living/carbon/human/human_armor.dm [hunk 2]
```diff
@@ -74,6 +74,22 @@
...
+			damage_flag = IMPALING
+		else if(sharpness & SHARP_POINTY)
+			damage_flag = PIERCING
+		else if(sharpness & SHARP_EDGED)
+			damage_flag = CUTTING
+
+	if(wear_suit && istype(wear_suit, /obj/item/clothing/suit/space/hardsuit/ms13/power_armor))
+		if((def_zone == BODY_ZONE_HEAD || def_zone == BODY_ZONE_PRECISE_EYES || def_zone == BODY_ZONE_PRECISE_MOUTH) && !wear_suit:helmettype)
+			return -damage
+		return wear_suit.get_integrity() - wear_suit.take_damage(damage, damage_type, damage_flag, 0, def_zone = def_zone)
```
Comment (Koshenko) [inline mojave/code/modules/mob/living/carbon/human/human_armor.dm]: Why return `-damage` when most calls to `damage_armor` are set to negative as well?  Is this just to account for line 91 where the return value could be positive?  Would it just be easier to return `damage` without setting it to negative and then for line 91 just adding a conditional to return a value of 0 if the damage doesn't get past the armor integrity?

**Review by Koshenko — DISMISSED** [review]

**Comment by ProfessorPopoff** [comment]
T45 helmet wasn't replaced. 
image

**Comment by ProfessorPopoff** [comment]
Some things I've noticed.

Right T45 arms are misnamed in the DMI (you already heard about this one)

Limb based damage/health seems to not work properly still. You can shoot their head and when it breaks, the entire set breaks. Not only that but the health examine text isn't working right.

image
image

"Lightly damaged" is putting it simply.


People bleeding when hit even though no damage was applied should probably be fixed.

**Comment by ProfessorPopoff** [comment]
Seemingly only with helmets, If you remove the helmet with the screwdriver and then get back into it, it'll give you a helmet anyways.

**Pull request merged**